crates/wasmparser/src/binary_reader.rs

```diff
@@ -2507,3 +2258,552 @@ impl fmt::Debug for BrTable<'_> {
         f.finish()
     }
 }
+
+/// A factory to construct [`Operator`] instances via the [`VisitOperator`] trait.
+pub struct OperatorFactory<'a> {
```

Comment:
I think it's ok to avoid this being public.

Reply:
True!